operations/model.go

```diff
@@ -609,16 +610,32 @@ func (s *ClientSettings) SetAutoUpgradeCLI() {
 	grip.Info("Evergreen CLI will be automatically updated and installed before each command if a more recent version is detected.")
 }
 
-// SetOAuthToken sets the OAuth token for authentication.
-func (s *ClientSettings) SetOAuthToken(ctx context.Context, comm client.Communicator) error {
-	token, path, err := comm.GetOAuthToken(ctx,
+func (s *ClientSettings) getOAuthToken(ctx context.Context, comm client.Communicator) (*oauth2.Token, string, error) {
+	return comm.GetOAuthToken(ctx,
 		s.OAuth.DoNotUseBrowser,
 		dex.WithIssuer(s.OAuth.Issuer),
 		dex.WithClientID(s.OAuth.ClientID),
 		dex.WithConnectorID(s.OAuth.ConnectorID),
 	)
+}
+
+// SetOAuthToken sets the OAuth token for authentication.
+func (s *ClientSettings) SetOAuthToken(ctx context.Context, comm client.Communicator) error {
+	token, path, err := s.getOAuthToken(ctx, comm)
 	if err != nil {
-		return errors.Wrap(err, "setting OAuth token")
+		// The auth library caches tokens in a file. Sometimes, the tokens are expired and
+		// we need to remove the file to get a new token.
+		if path != "" {
+			if delErr := os.Remove(path); delErr != nil && !os.IsNotExist(delErr) {
```

Comment:
nit: you could use `os.RemoveAll` since it returns no error if the file is nonexistent.